gammapy/cube/sherpa_.py

```diff
@@ -113,6 +113,143 @@ def notice(self, x0lo=None, x0hi=None, x1lo=None, x1hi=None, x2lo=None, x2hi=Non
                         ignore)
 
 
+class Data3DInt(DataND):
+    "3-D integrated data set"
+
+    def _set_mask(self, val):
+        DataND._set_mask(self, val)
+        try:
+            self._x0lo = self.apply_filter(self.x0lo)
+            self._x0hi = self.apply_filter(self.x0hi)
+            self._x1lo = self.apply_filter(self.x1lo)
+            self._x1hi = self.apply_filter(self.x1hi)
+            self._x2lo = self.apply_filter(self.x2lo)
+            self._x2hi = self.apply_filter(self.x2hi)
+        except DataErr:
+            self._x0lo = self.x0lo
+            self._x1lo = self.x1lo
+            self._x0hi = self.x0hi
+            self._x1hi = self.x1hi
+            self._x2hi = self.x2hi
+            self._x2hi = self.x2hi
+
+    mask = property(DataND._get_mask, _set_mask,
+                    doc='Mask array for dependent variable')
+
+    def __init__(self, name, x0lo, x1lo, x2lo, x0hi, x1hi, x2hi, y, shape=None,
+                 staterror=None, syserror=None):
+        self._x0lo = x0lo
+        self._x1lo = x1lo
+        self._x2lo = x2lo
+        self._x0hi = x0hi
+        self._x1hi = x1hi
+        self._x2hi = x2hi
+        BaseData.__init__(self)
+
+    def get_indep(self, filter=False):
+        filter = bool_cast(filter)
+        if filter:
+            return (self._x0lo, self._x1lo, self._x2lo, self._x0hi, self._x1hi, self._x2hi)
+        return (self.x0lo, self.x1lo, self.x2lo, self.x0hi, self.x1hi, self.x2hi)
+
+    def get_x0(self, filter=False):
+        indep = self.get_indep(filter)
+        return (indep[0] + indep[3]) / 2.0
+
+    def get_x1(self, filter=False):
+        indep = self.get_indep(filter)
+        return (indep[1] + indep[4]) / 2.0
+
+    def get_x2(self, filter=False):
+        indep = self.get_indep(filter)
+        return (indep[2] + indep[5]) / 2.0
+
+    def notice(self, x0lo=None, x0hi=None, x1lo=None, x1hi=None, x2lo=None, x2hi=None, ignore=False):
+        BaseData.notice(self, (x0lo, x1lo, x2hi),
+                        (x0hi, x1hi, x2hi), self.get_indep(), ignore)
+
+
+class CombinedModel3D(ArithmeticModel):
```

Comment:
This is probably a copy and paste error...the same class is at the bottom of the file. Could you remove it?